lib/RoadHandler.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * TerrainHandler.h, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+
+#pragma once
+
+#include <vcmi/EntityService.h>
+#include <vcmi/Entity.h>
+#include "GameConstants.h"
+#include "IHandlerBase.h"
+
+VCMI_LIB_NAMESPACE_BEGIN
+
+class DLL_LINKAGE RoadType : public EntityT<RoadId>
+{
+	friend class RoadTypeHandler;
+	std::string identifier;
+	RoadId id;
+
+	const std::string & getName() const override { return identifier;}
+public:
+	int32_t getIndex() const override { return id.getNum(); }
+	int32_t getIconIndex() const override { return 0; }
+	const std::string & getJsonKey() const override { return identifier;}
+	void registerIcons(const IconRegistar & cb) const override {}
+	RoadId getId() const override { return id;}
+	void updateFrom(const JsonNode & data) {};
+
+	std::string getNameTextID() const;
+	std::string getNameTranslated() const;
+
+	std::string tilesFilename;
+	std::string shortIdentifier;
+	ui8 movementCost;
+
+	RoadType();
+
+	template <typename Handler> void serialize(Handler& h, const int version)
+	{
+		h & tilesFilename;
```

Comment:
Backward compatibility is broken again? Do not forget to change serialization version if yes